### PR TITLE
implemented pending bucket deletions test harness using H2

### PR DESCRIPTION
**Pull request by dvoet**

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/PendingBucketDeletionComponent.scala [hunk 1]
```diff
@@ -8,7 +8,7 @@ trait PendingBucketDeletionComponent {
   import driver.api._
 
   class PendingBucketDeletions(tag: Tag) extends Table[PendingBucketDeletion](tag, "BUCKET_DELETION") {
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/PendingBucketDeletionComponent.scala]: I'm not a fan of this naming convention: too close to the `PendingBucketDeletion` case class.  What about `PendingBucketDeletionTable` for the `Table` and `pendingBucketDeletionQuery` for the `TableQuery`?

Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/PendingBucketDeletionComponent.scala]: sure

File: src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/TestDriverComponent.scala [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import slick.driver.H2Driver.api._
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, Future}
+
+/**
+ * Created by dvoet on 2/3/16.
+ */
+trait TestDriverComponent extends DriverComponent {
+  override implicit val executionContext = scala.concurrent.ExecutionContext.Implicits.global
```
Comment (jmthibault79) [inline src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/TestDriverComponent.scala]: Could use the `TestExecutionContext` here.

File: src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/TestDriverComponent.scala [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  val databaseConfig: DatabaseConfig[JdbcProfile] = DatabaseConfig.forConfig[JdbcProfile]("h2mem1")
+  override val driver: JdbcProfile = databaseConfig.driver
+  val database = databaseConfig.db
+
+  protected def runAndWait[R](action: DBIOAction[R, _ <: NoStream, _ <: Effect], duration: Duration = Duration.Inf): R = {
```
Comment (jmthibault79) [inline src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/TestDriverComponent.scala]: Inf makes me uncomfortable.  Could we make it a different unreasonably long duration, say 1 minute?

**Comment by jmthibault79** [comment]
This is pretty great.  Thanks!

**Comment by jmthibault79** [comment]
:+1:

**Pull request merged**